### PR TITLE
fetch: ignore force-mirror feature when try_mirrors is false

### PR DESCRIPTION
**Pull request by floppym**
When fetching layout.conf, we call fetch(try_mirrors=False) to force direct use of the primary URI being passed and ignore mirrors.

The force-mirror feature causes fetch to ignore the primary URI.

Ignoring both the primary URI and mirrors leads to an unfetchable file. Give precedence to the try_mirrors parameter since we really only set it to False when fetching layout.conf from a mirror.

Bug: https://bugs.gentoo.org/877793

### CODE REVIEW

**Review by thesamesam — APPROVED** [review]
I find `fetch.py` kind of hard to follow, but this sounds right. I'd like mgorny to look ideally given he implemented the GLEP 75 bits.

**Review by mgorny — APPROVED** [review]
LGTM. I was a little worried this could affect `RESTRICT=mirror`-ed packages but now I see that the logic for detecting that is inlined ugly into `fetch()`, sigh.

**Comment by floppym** [comment]
Yeah, fetch could definitely use a refactor.
